### PR TITLE
Updating search branch to master branch

### PR DESCRIPTION
**Pull request by joshuabello2550**

### CODE REVIEW

File: app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java [hunk 1]
```diff
@@ -88,9 +88,7 @@ private void setTakePictureOnClickListener() {
         btnTakePicture.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                // create Intent to take a picture and return control to the calling application
                 Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
-                // Create a File reference for future access
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java:93]: It's create that you are applying my feedback :)

File: app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java [hunk 2]
```diff
@@ -152,16 +146,18 @@ private void setAddVehicleOnClickListener() {
     }
 
     private void getPlaceId(String streetAddress, String city, String state) {
-        String base_url = "https://maps.googleapis.com/maps/api/geocode/json?";
-        String address = streetAddress + " " + city + " " + state;
+        final String GOOGLE_GEOCODING_API_BASE_URL = "https://maps.googleapis.com/maps/api/geocode/json?";
+        final String GOOGLE_GEOCODING_API_PARAMETER_KEY = "key";
+        final String GOOGLE_GEOCODING_API_PARAMETER_ADDRESS = "address";
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java:151]: You would want to move this to class level. BTW, 

public class PhysicalConstants {
    private PhysicalConstants() { }  // Prevents instantiation

    private static final double AVOGADROS_NUMBER   = 6.02214199e23;
    private static final double BOLTZMANN_CONSTANT = 1.3806503e-23;
    private static final double ELECTRON_MASS      = 9.10938188e-31;
}

File: app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java [hunk 3]
```diff
@@ -54,32 +54,30 @@ public ExploreAdapter(Fragment fragment, List<Vehicle> vehicles, MainActivity ac
...
-        mVehicles.addAll(allVehicles);
+        vehicles.addAll(allVehicles);
         notifyDataSetChanged();
     }
 
     public void clear() {
-        mVehicles.clear();
+        vehicles.clear();
         notifyDataSetChanged();
     }
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java:83]: Have we considered combining them into one method `setVehicles()`?

File: app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java [hunk 4]
```diff
@@ -123,19 +128,23 @@ public void setValues(Vehicle vehicle) {
         }
 
         private void getDistanceFromUser(Vehicle vehicle, String vehiclePlaceId, String userLocation) {
-            String baseUrl = "https://maps.googleapis.com/maps/api/distancematrix/json?";
+            final String GOOGLE_DISTANCE_MATRIX_API_BASE_URL = "https://maps.googleapis.com/maps/api/distancematrix/json?";
+            final String GOOGLE_DISTANCE_MATRIX_API_PARAMETER_DESTINATIONS = "destinations";
+            final String GOOGLE_DISTANCE_MATRIX_API_PARAMETER_ORIGINS = "origins";
+            final String GOOGLE_DISTANCE_MATRIX_API_PARAMETER_KEY = "key";
+            final String GOOGLE_DISTANCE_MATRIX_API_PARAMETER_UNITS = "units";
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java:135]: ^

File: app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java [hunk 5]
```diff
@@ -181,21 +190,19 @@ public void onClick(View v) {
...
+                        setSaveButtonClickedStyle(ibSave);
                     }
                     currentUser.setSavedVehicles(listSavedVehicles);
                     currentUser.saveInBackground();
                 }
             });
         }
 
-        private void setSaveBtnClicked(ImageButton ibSave) {
+        private void setSaveButtonClickedStyle(ImageButton ibSave) {
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/adapters/ExploreAdapter.java:201]: Don't you have ibSave as your instance variable?

File: app/src/main/java/com/example/xpvehicles/adapters/RentingVehiclesAdapter.java [hunk 6]
```diff
@@ -136,23 +134,28 @@ public void setValues(RentVehicle rentvehicle) throws ParseException{
         }
 
         private void setStatusColor(String status) {
+            final String STATUS_PENDING_APPROVAL = "pending approval";
+            final String STATUS_APPROVED = "approved";
+            final String STATUS_PENDING_DENIED = "denied";
```
Comment (leixiangwu) [inline app/src/main/java/com/example/xpvehicles/adapters/RentingVehiclesAdapter.java:139]: You can use enum to represent these states

File: app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java (hunk 2 above)
Comment (joshuabello2550) [reply app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java:151]: What is the point of preventing instantiation with "private PhysicalConstants() { } // Prevents instantiation"

Comment (joshuabello2550) [reply app/src/main/java/com/example/xpvehicles/activities/AddVehicleActivity.java:151]: Should I still move this to the class level, even those these constants are only used in this function

File: app/src/main/java/com/example/xpvehicles/adapters/RentingVehiclesAdapter.java (hunk 6 above)
Comment (joshuabello2550) [reply app/src/main/java/com/example/xpvehicles/adapters/RentingVehiclesAdapter.java:139]: I created this class, but am unsure how to access the values I saved

package com.example.xpvehicles.Miscellaneous;

import androidx.annotation.NonNull;

public enum RentingStatus {
    PENDING_APPROVAL ("pending status"),
    READY ("approved"),
    DENIED ("denied"),
    ;

    private String theStatus;

    RentingStatus(String s) {
        theStatus =  s;
    }

    @NonNull
    @Override
    public String toString() {
        return theStatus;
    }
}

**Pull request merged**